unstable/logging/logging.go

```diff
@@ -106,13 +179,177 @@ func (l *Host[L]) f(severity diag.Severity, msg string) {
 	glog.V(9).Infof("[%s]: %q", sev, msg)
 }
 
-func (l *Host[L]) Status() L {
+func (l *host[L]) Status() L {
 	copy := *l
 	copy.status = true
 	return l.mkStatus(&copy)
 }
 
-func (l *Host[L]) Debug(msg string) { l.f(diag.Debug, msg) }
-func (l *Host[L]) Info(msg string)  { l.f(diag.Info, msg) }
-func (l *Host[L]) Warn(msg string)  { l.f(diag.Warning, msg) }
-func (l *Host[L]) Error(msg string) { l.f(diag.Error, msg) }
+func (l *host[L]) StatusUntyped() any {
+	return l.Status()
+}
+
+func (l *host[L]) Debug(msg string) { l.f(diag.Debug, msg) }
+func (l *host[L]) Info(msg string)  { l.f(diag.Info, msg) }
+func (l *host[L]) Warn(msg string)  { l.f(diag.Warning, msg) }
+func (l *host[L]) Error(msg string) { l.f(diag.Error, msg) }
+
+// Directs any logs written using the tflog API in the given Context to the given output.
+func setupRootLoggers(ctx context.Context, output io.Writer) context.Context {
+	desiredLevel := parseTfLogEnvVar()
+	sdkLoggerOptions := makeLoggerOptions("sdk", desiredLevel, output)
+	ctx = context.WithValue(ctx, sdkKey, hclog.New(sdkLoggerOptions))
+	ctx = context.WithValue(ctx, sdkOptionsKey, sdkLoggerOptions)
+	providerLoggerOptions := makeLoggerOptions("provider", desiredLevel, output)
+	ctx = context.WithValue(ctx, providerKey, hclog.New(providerLoggerOptions))
+	ctx = context.WithValue(ctx, providerOptionsKey, providerLoggerOptions)
+	return ctx
+}
+
+// Choose the default level carefully: logs at this level or higher (more severe) will be shown to the user of Pulumi
+// CLI directly by default. Experimentally it seems that Info is too verbose, for example Cloudflare provider emits
+// routine authentication messages at INFO level.
```

Comment:
What does `terraform` do here? It feels like we should mirror what they do.

Reply:
This is an interesting question. I'd call it out of scope for this PR, note that this code is NOT net new but it's moving around the code that used to do exactly this for Plugin Framework and installing it in the SDKv2 context. I'd also say we're not bound by what TF does necessarily, though it can inform it. 

There's also https://github.com/pulumi/pulumi-terraform-bridge/issues/97 that has some different ideas of tying things to the debug flag.  

Actually coming to think of it, because of https://github.com/pulumi/pulumi-terraform-bridge/pull/1537/commits/dc75cc06ebd4f0cd8d3a1ac6e81e65bc10169a91 there will be an observable change here that WARNING level logs will now start showing up that didn't show up previously, even in PF case that's not suppose to be changing. I'm open to tightening the default level to ERROR so that doesn't happen. It's quite possible that if we do not do this, we'll receive requests to close off un-actionable TF warning logs  right after we ship this. Thoughts?